app/src/main/java/com/github/polybooks/utils/Futures.kt

```diff
@@ -0,0 +1,13 @@
+package com.github.polybooks.utils
+
+import java.util.*
+import java.util.concurrent.CompletableFuture
+
+fun <T> listOfFuture2FutureOfList(listOfFuture : List<CompletableFuture<T>>) : CompletableFuture<List<T>> {
+    val futureOfEmptyList = CompletableFuture.completedFuture<List<T>>(Collections.emptyList())
+    return listOfFuture.fold(futureOfEmptyList) { acc, curr ->
+            acc.thenCombine(curr) { listAcc, currElem ->
+                listAcc.plus(currElem)
+            }
+        }
+}
```

Comment:
Have you tested this function? It seems strange to me that line 7 works. You are created an already completed future with an empty list inside (the important part is *already completed*). Then, in the return statement, you basically populate this future with values found in the list of futures (the parameter). So I have two questions:
- Is it really legal in kotlin (all the other languages I know don't allow that) to modify the value contained in a *completed* future?
- what if the list of future contains elements (aka. futures) that are not resolved nor pending (aka. rejected)?

Reply:
Yes there are tests specifically for this function.
Why this works is because I am not modifying the content of a future, I am combining two futures into a new one. With a new list.
If the list contains a future that is rejected, the result is a future that is rejected.